### PR TITLE
fix(`crosschain`): remove confirmation mode from outbound and inbound digest

### PR DESCRIPTION
**Pull request by lumtis**
# Description

Closes https://github.com/zeta-chain/node/issues/4106

## Summary by CodeRabbit

- Bug Fixes
  - Standardized digest computation for inbound and outbound vote messages to be independent of confirmation mode, ensuring consistent hashes and improving reliability of cross-chain message verification.

- Tests
  - Updated test cases to confirm that changing the confirmation mode does not affect digest results.

- Documentation
  - Added changelog entry under Unreleased > Fixes noting the removal of confirmation mode influence from digest calculations.

### CODE REVIEW

File: x/crosschain/types/message_vote_inbound.go [hunk 1]
```diff
@@ -162,6 +162,7 @@ func (msg *MsgVoteInbound) Digest() string {
 	m := *msg
 	m.Creator = ""
 	m.InboundBlockHeight = 0
+	m.ConfirmationMode = ConfirmationMode_SAFE
```
Comment (kingpinXD) [inline x/crosschain/types/message_vote_inbound.go:165]: Replaying old transactions with ConfirmationMode_FAST would now generate a new index , but we should still be protected since the finalization index would exist . 
Please conisder adding a unit test to verify that .

Comment (kingpinXD) [reply x/crosschain/types/message_vote_inbound.go:165]: I don't think the ballot index would be the same for an inbound which was previously created with `ConfirmationMode_FAST`. 
In cases where we end up replaying a block , or creating a tracker .

But I belive the finalization index would still protect us , feel free to resolve the comment if you still  think the test is unnecessary

Comment (lumtis) [reply x/crosschain/types/message_vote_inbound.go:165]: >I don't think the ballot/cctx index would be the same for an inbound which was previously created with ConfirmationMode_FAST.
In cases where we end up replaying a block , or creating a tracker .

Before this upgrade we will have the patch that enforce all inbound to use SAFE so we will not have the index issue during the transition if this is the question

>Using only one single digest definitely fixes the issue. Looks like a fix in msg_server_vote_inbound_tx.go is also a small change (maybe I'm wrong?).
I was able to replicate it in a branch replicate-double-spent and start-e2e-test will trigger the edge case in modified E2E TestETHDeposit.
Feel free to resolve my comment if you think allowing two digests is not a easy fix because I might be wrong.

We should not add it in the digest even if it's fixed outside. The digest should be based on inbound data itself, not how it gets observed

**Review by kingpinXD — APPROVED** [review]

**Review by ws4charlie — APPROVED** [review]
